drivers/sensor/lis3mdl/Kconfig

```diff
@@ -39,10 +39,9 @@ config LIS3MDL_I2C_MASTER_DEV_NAME
 	  Specify the device name of the I2C master device to which LIS3MDL is
 	  connected.
 
-choice
+choice LIS3MDL_TRIGGER_MODE
 	prompt "Trigger mode"
 	depends on LIS3MDL
-	default LIS3MDL_TRIGGER_GLOBAL_THREAD
```

Comment:
any reason to remove the default?

Reply:
Yes, I think I explained this in the commit. This is required (at least I couldn't do otherwise) to be able to define a value in board Kconfig.defconfig

Reply:
The commit message explains why you added a name to the choice, not why you removed the default.

Reply:
This is right! 
So actually this is the combination of 2 (remove default and provide name to the choice) which was required.
Unfortunately, this commit has already been merged in arm branch, so I won't be able to make it clearer.
(And current commit should be remove when rebasing)